Src/IronPython.Modules/_ctypes/Field.cs

```diff
@@ -42,51 +42,39 @@ public sealed class Field : PythonTypeDataSlot, ICodeFormattable {
                 _index = index;
                 _fieldName = fieldName;
 
-                // if the number of bits is the full type width, we don't
-                // need to do any masking anywhere, we may want to revisit this
-                // if the __repr__ is an issue
-                if (bits != null && bits.Value != (_fieldType.Size * 8)) {
+                if (bits != null) {
                     _bits = bits.Value;
                     _bitsOffset = bitOffset.Value;
+                    if (((SimpleType)_fieldType)._swap) {
```

Comment:
I guess it's guaranteed to be a `SimpleType` if `bits is not null`?

Reply:
Yes, and the code depended on this constraint already before but it was not that visible (see `IsSignedType` and its use). I'm gonna drop in an explicit debug assert here.